### PR TITLE
Clarify module choices in quickstart docs

### PR DESCRIPTION
**Pull request by armanbilge**
Follow-up to https://github.com/http4s/http4s/pull/6963#discussion_r1100664745. Adapted from similar documentation in Cats Effect.
https://github.com/typelevel/cats-effect#getting-started

### CODE REVIEW

File: docs/docs/quickstart.md [hunk 1]
```diff
@@ -2,21 +2,24 @@
 
 ## Dependencies
 
-http4s is available for Scala 2.12, 2.13, and 3.
-
-You can add http4s to your build by adding its modules to your dependencies in `build.sbt`.
-Here are some common dependencies, more are available on the [integrations page][integrations].
+http4s is available for Scala 2.12, 2.13, and 3. You can add http4s to your build by adding its modules to `libraryDependencies` in your `build.sbt`.
```
Comment (danicheg) [inline docs/docs/quickstart.md:5]: Just wondering, is Scala 3 users so familiar with brand-new behavior of absence of forward compatibility? Shouldn't we try to cut down the cognitive load for newcomers by taking the process debt of updating docs when a new minor version of Scala 3 is dropped?

**Review by danicheg — APPROVED** [review]

File: docs/docs/quickstart.md (hunk 1 above)
Comment (armanbilge) [reply docs/docs/quickstart.md:5]: Yeah, it's a good point, I have seen confusion here. So I will change it to say `3.2+`. Maybe when 3.3 drops, and it is an LTS, we will stay on it for a while ... but remains to be seen.

Also, Scala 2.13 might be dropping forward-compatibility as well soon, to add to the confusion 😅 
- https://github.com/scala/improvement-proposals/pull/54

**Comment by danicheg** [comment]
@armanbilge thanks for iterating on this!

File: docs/docs/quickstart.md [hunk 2]
```diff
@@ -2,21 +2,24 @@
...
   "org.http4s" %% "http4s-ember-client" % http4sVersion,
   "org.http4s" %% "http4s-ember-server" % http4sVersion,
   "org.http4s" %% "http4s-dsl"          % http4sVersion,
+)
+```
+
+This brings in the [Ember HTTP client and server implementation][ember] as well as the the [routing DSL][dsl]. This is _most likely_ what you want. Other [integration modules][integrations] are available as well, such as for working with [JSON], or using alternative client/server implementations, such as [Netty].
 
-  // for core classes and traits, e.g. `Client[F]`
+Depending on your usecase, you may want to consider one of several other http4s modules. For example, if you are writing a library, you probably want to develop it against the generic http4s interfaces, without hard-coding a particular implementation, such as Ember.
```
Comment (rossabaker) [inline docs/docs/quickstart.md:19]: It might also be worth mentioning that some of these others may be versioned separately.

**Comment by armanbilge** [comment]
Always happy to see more iterations :)

**Pull request merged**